### PR TITLE
[inductor] Make graph.py pass mypy

### PR DESCRIPTION
**Pull request by haashah**
Apply fixes suggested by mypy to the file

* torch/_inductor/graph.py

Addressing following errors:

graph.py:111: error: Argument 1 to "convert_shape_to_inductor" has incompatible type "Size"; expected "list[int | SymInt]"  [arg-type]
graph.py:112: error: Argument 1 to "convert_shape_to_inductor" has incompatible type "tuple[int, ...]"; expected "list[int | SymInt]"  [arg-type]
graph.py:515: error: "Tensor" has no attribute "_has_symbolic_sizes_strides"  [attr-defined]
graph.py:936: error: "IRNode" has no attribute "get_name"; maybe "get_numel"?  [attr-defined]

See #105230

Fixes #105230 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -135,7 +135,9 @@ def next_power_of_2(n):
     return n
 
 
-def convert_shape_to_inductor(lst: List[Union[int, torch.SymInt]]) -> List[sympy.Expr]:
+def convert_shape_to_inductor(
+    lst: Union[List[Union[int, torch.SymInt]], torch.Size, tuple[int, ...]]
```
Comment (Skylion007) [inline torch/_inductor/utils.py:139]: Wouldn't a tuple of SymInt also be acceptable to pass? We should expand the type to include that.

File: torch/_inductor/graph.py [hunk 2]
```diff
@@ -172,8 +172,8 @@ def __init__(
         self._shape_env = shape_env
         self.sizevars = SizeVarAllocator(shape_env)
         self.graph_inputs: Dict[str, TensorBox] = {}
-        self.graph_inputs_original: Dict[str, InputBuffer] = {}
-        self.graph_outputs: Optional[List[ir.IRNode]] = None
+        self.graph_inputs_original: Dict[str, Union[InputBuffer, ir.StorageBox]] = {}
+        self.graph_outputs: Optional[List[Union[ir.StorageBox, ir.InputBuffer]]] = None
         self.device_types: Set[str] = set()
         self.device_idxs: Set[int] = set()
         self.cuda = False
```
Comment (Skylion007) [inline torch/_inductor/graph.py:179]: This should be typed as a bool, right?

**Review by Skylion007 — COMMENTED** [review]
Some comments

**Comment by haashah** [comment]
pytorchbot label "topic: not user facing"

File: torch/_inductor/utils.py (hunk 1 above)
Comment (haashah) [reply torch/_inductor/utils.py:139]: original error states:

graph.py:111: error: Argument 1 to "convert_shape_to_inductor" has incompatible type "Size"; expected "list[int | SymInt]"  [arg-type]
graph.py:112: error: Argument 1 to "convert_shape_to_inductor" has incompatible type "tuple[int, ...]"; expected "list[int | SymInt]"  [arg-type]

Comment (haashah) [reply torch/_inductor/utils.py:139]: so doesn't look like a tuple of SymInt will satisfy the usage in graph.py

File: torch/_inductor/graph.py (hunk 2 above)
Comment (haashah) [reply torch/_inductor/graph.py:179]: right will add

**Comment by haashah** [comment]
I see graph.py is unnessary to type fix as its already part of the linter:
https://github.com/pytorch/pytorch/blob/dc19f8a6b5d051cd01a1fad6bfadb898ba466f6e/.lintrunner.toml#L182
Will close this PR and pickup another file from https://github.com/pytorch/pytorch/issues/105230

**Pull request closed without merging**